### PR TITLE
Refactor report submit form

### LINKED ISSUE #1135: Refactor ReportSubmitForm to use new Form component

### PR DESCRIPTION
**Pull request by cailyncodes**
Should close #1135 

My self review will come later tonight or tomorrow morning. Feel free to hold off reviewing until then

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/ChildInfo/NewForm.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ export const NewForm: React.FC<SectionProps> = ({
 
 	return (
 		<Form<Enrollment>
-			className="ChildInfoForm"
+			className="ChildInfoForm usa-form"
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Enrollment/_sections/ChildInfo/NewForm.tsx:119]: why is this needed? Shouldn't this already be on the div that the `Form` component creates?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/C4KRevenue.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	c4kFundedEnrollments.forEach((enrollment) => {
+		const childId = enrollment.childId;
+		if (childIds.indexOf(childId) < 0) {
+			childIds.push(childId);
+		}
+	});
+	const care4KidsCount = childIds.length;
+
+	return (
+		<TextInput
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/C4KRevenue.tsx:51]: why not form field?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/FamilyFeesRevenue.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+export const FamilyFeesRevenueField: React.FC<FamilyFeesRevenueFieldProps> = ({
+	disabled,
+	error,
+	errorAlertState,
+}) => {
+	const { data, dataDriller, updateData } = useGenericContext<CdcReport>(FormContext);
+	const familyFeesRevenue = data.familyFeesRevenue;
+
+	return (
+		<TextInput
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/FamilyFeesRevenue.tsx:34]: why not form field?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx [hunk 4]
```diff
@@ -46,16 +54,10 @@ interface UtilizationTableRow {
 	};
 }
 
-type UtilizationTableProps = {
-	report: CdcReport;
-	timeSplitUtilizations: FundingTimeSplitUtilization[];
-};
+export default function UtilizationTable() {
+	// Pass in the state variable from reportSubmitForm that contains the utilization values for this report (either default values or user-defined)): React.ReactElement<UtilizationTableProps> {
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx:58]: pass in ?

Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx:58]: ah i see. it is no longer necessary to pass in the timeSplitUtilizations b/c they are updated via a formField on the same report object, and thus are set on the report in here?

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/RetroactiveC4KRevenue.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+type RetroactiveC4KRevenueProps = {
+	disabled: boolean;
+};
+/**
+ * Component for entering whether the entered c4k revenue includes retroactive
+ * payments.
+ */
+export const RetroactiveC4KRevenue: React.FC<RetroactiveC4KRevenueProps> = ({ disabled }) => {
+	return (
+		<CheckboxGroup
```
Comment (jlhogan) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/RetroactiveC4KRevenue.tsx:16]: also, why a checkboxGroup?

File: src/Hedwig/ClientApp/src/containers/Enrollment/_sections/ChildInfo/NewForm.tsx (hunk 1 above)
Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Enrollment/_sections/ChildInfo/NewForm.tsx:119]: I had to get rid of that so that the usage in `ReportSubmitForm` would work. `usa-form` makes a max width on the container. The `ReportSubmitForm` comps had a fully expanded section followed by a condensed section

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/C4KRevenue.tsx (hunk 2 above)
Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/C4KRevenue.tsx:51]: Because I had to use the `useGenericContext` manually for the c4k child count, I think I just went with this approach. (Also the onBlur might have been something, specifically the setting of `event.target.value`)

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/FamilyFeesRevenue.tsx (hunk 3 above)
Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/FamilyFeesRevenue.tsx:34]: Same thing about onBlur

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/RetroactiveC4KRevenue.tsx (hunk 5 above)
Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/RetroactiveC4KRevenue.tsx:16]: I needed the legend to make design comps

Comment (cailyncodes) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/ReportSubmitFields/RetroactiveC4KRevenue.tsx:16]: Perhaps you open PR would address that though
